CIP-????/README.md

```diff
@@ -0,0 +1,530 @@
+would be executed against the full transaction. This requirement enables the current smart contract
+interface to remain mostly untouched. The flow is essentially:
+
+1. Deterministically derive the whole transaction from the list of pieces.
+2. Generate the script context for the whole transaction.
+3. Evaluate the smart contract using this script context.
+
+When combining the transaction pieces, the required executions (ie, script + datum + redeemer) can
+be built up as well. Duplicate executions should be removed. For example, if two transaction pieces
+look to spend the same smart contract input using the same redeemer, there is no reason to evaluate
+this scenario twice since they are guaranteed to have the same result.
+
+### Transaction Fee and Collateral
+
+Both the fee and collateral fields of the transaction body can be set in any piece, but only the
+aggregator needs to set this (in their final piece that is used to finish balancing the batch). They
+have all of the pieces that will go into the batch which means they can deterministically calculate
+the required fee and collateral for the batch they are creating. Their final piece can be updated
+after determining the required fee and collateral.
+
+By having the aggregator put up the collateral, they are naturally incentivized not to include any
+transaction pieces with smart contract executions that will fail. They can know whether the smart
+contract execution will fail ahead of submission due to the determinism of the eUTxO model being
+preserved, despite the transaction body being broken into pieces. The same incentives apply for the
+transaction fee since the aggregator will want to minimize their own costs, and therefore, won't
+waste execution resources.
+
+### Transaction Piece Monoid Instance
+
+The monoid instance for transaction pieces defines the rules for combining the pieces. In order to
+be able to deterministically derive the full transaction, this instance must be standardized. The
+monoid instance requires answering two questions:
+
+1. What is an empty transaction piece?
+2. How are two transaction pieces combined?
+
+The first question is simple to answer: an empty transaction piece is an empty transaction body.
+Users fill out as much information as they wish when creating their transaction piece. The rest of
+the fields can remain empty.
+
+> [!IMPORTANT]
+> It doesn't really make sense to create a transaction piece with an input but no output, or an
+> output but no input. Therefore, these fields can remain required fields. The fee is also required,
+> but that can be set to zero in the pieces not paying the fee. All other fields in the transaction
+> body are already optional.
+
+The second question requires more consideration, but as a general rule, as long as there are no
+incompatibilities between transaction pieces, they can be combined into the full transaction. Any
+transaction that contains incompatible pieces should be rejected.
```

Comment:
Any two transactions that have plutus scripts would be immediately incompatible.

Imagine a script that relies for there to be only a single input being spent or a single output being produced. As soon as you combine two transactions together they no longer validate. However, you can't know that until you try to combine them and run the scripts

There is another fundamental problem here. Imagine those inputs in two separate transactions are both locked by scripts. Those two transactions both provide script integrity hash that have potentially been signed. But whenever you combine those two transactions together you must change their redeemers and pass those redeemer modified in the script context to plutus scripts, which is totally screwing with the data that have been signed.

So, combining multiple pieces into one transaction idea is definitely out of the window